shard-operator/src/main/resources/templates/service-monitor.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: ""
+  labels:
+    app.kubernetes.io/managed-by: bridge-fleet-shard-operator # used for filtering of Deployments created by the controller
+    app.kubernetes.io/created-by: bridge-fleet-shard-operator # Specify the operator
+    app.kubernetes.io/component: ingress # used to specify the component
```

Comment:
This implementation is already flexible enough to remove this `app.kubernetes.io/component: ingress` and set it programmatically: we can reuse everything for the `executor` services. 
We don't have to do it now.. we can do it in the PR for the `executor` part